src/XRP/xrp-currency.ts

```diff
@@ -0,0 +1,20 @@
+import { Currency } from '../generated/web/org/xrpl/rpc/v1/amount_pb'
+
+/**
+ * An issued currency on the XRP LEdger
+ * - SeeAlso: https://xrpl.org/currency-formats.html#currency-codes
+ */
+export default class XRPCurrency {
+  public name: string
+
+  public code: string | Uint8Array
+
+  public static from(currency: Currency): XRPCurrency | undefined {
+    return new XRPCurrency(currency)
+  }
+
+  private constructor(currency: Currency) {
```

Comment:
Typescript lets you create readonly iVars + bindings in one constructor.  So you can pretty drastically reduce boiler plate code here:
```
export class Foo {
  private constructor(public readonly accountID: string, public readonly assetCode: string, ...)

  static from(getBalanceResponse: GetBalanceResponse) {
      return Foo(getBalanceResponse.getAccountID(), ...)
   }
}
```

https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties